### ISSUE TITLE
Better folder naming conventions?

### ISSUE THREAD

**Issue by cyberglot** [heliaxdev/juvix#734]
I've noticed that some folders have different kind of names, and we don't seem to be following any conventions. Maybe it would be nice to set on something before it becomes out of hand.

At top level, we have `src`, `app` and `library`. I suggest we either have it all fully spelled out or the abbreviation. In Another case is within the `library` folder, we have `Frontend`, `Backend`, `Translate`. I suggest we go for nouns, making `Translate` into `Translation`.
Lastly, we have cases where some folders are prefixed and some aren't, like `FrontendContextualise` and `Desugar`. Maybe `Contextualisation` and `Desugaring` would be better.

**Comment by andy-morris** [comment]
I also think that maybe the folder names should be lowercase, like the names of the libraries inside them. (Or capitalise the library names, I suppose.)

**Comment by cwgoes** [comment]
Let's have frontend import core instead of `FromFrontend`.

**Comment by cwgoes** [comment]
And let's definitely avoid e.g. `FrontendContextualise`, if we have a name like that it probably means we did something wrong in the folder structure.

**Comment by cwgoes** [comment]
Also, we should use American English spelling everywhere.

**Comment by cwgoes** [comment]
Nouns it is.

**Comment by fdedden** [comment]
> I also think that maybe the folder names should be lowercase, like the names of the libraries inside them. (Or capitalise the library names, I suppose.)

Yes I think the folders directly under `library/` should be lowercase; they are not referring to a module and therefore don't need to be uppercase.

**Comment by cwgoes** [comment]
Part of the idealised pipeline / rearchitecture discussions.

Also, we need to fix the library module names, which have inconsistent capitalisation.
